bindings/cs/Scripts/build.cake

```diff
@@ -0,0 +1,218 @@
+        CopyFile(
+            libmongocryptAllDirectory.Combine("macos").Combine("nocrypto").Combine("lib").CombineWithFilePath("libmongocrypt.dylib"),
+            downloadedMongocryptDirectory.CombineWithFilePath("libmongocrypt.dylib"));
+        CopyDirectory(downloadedMongocryptDirectory, libmongocryptRelWithDebInfoDirectory);
+    });
+
+Task("Tests")
+    .IsDependentOn("Prepare")
+    .DoesForEach(
+    () => 
+    {
+        var monikersDetails = new List<(string Moniker, string Bitness)>
+        {
+            { ("net452", "x64") },
+            { ("netcoreapp1.1", "x64") },
+            { ("netcoreapp2.1", "x64") },
+            { ("netcoreapp3.0", "x64") },
+            { ("net50", "x64") }
+        };
+        return monikersDetails;
+    },
+    (monikerInfo) =>
+    {
+        Information($"Test running {monikerInfo.Moniker}..");
+        var settings = new DotNetCoreTestSettings
+        {
+            Configuration = configuration,
+            Framework = monikerInfo.Moniker,
+            ArgumentCustomization = args => args.Append($"-- RunConfiguration.TargetPlatform={monikerInfo.Bitness}")
+        };
+        var projectFullPath = libmongocryptTestsSolutionDirectory.CombineWithFilePath("MongoDB.Libmongocrypt.Test.csproj").FullPath;
+        Information(projectFullPath);
+        DotNetCoreTest(
+            projectFullPath,
+            settings
+        );
+    })
+    .DeferOnError();
+    
+Task("CreatePackage")
+    .IsDependentOn("Tests")
+    .Does(() =>
+    {
+        var projectFullPath = libmongocryptSolutionFile.FullPath;
+        var settings = new DotNetCorePackSettings
+        {
+            Configuration = configuration,
+            OutputDirectory = artifactsDirectory
+        };
+        DotNetCorePack(
+            projectFullPath,
+            settings);            
+    });
+    
+Task("NugetPush")
+    .Does(() =>
+    {
+        ThrowIfLocalRelease();
+
+        var nugetApi = GetSettingValue("NugetApiKey", null);
+        var packageFilePath = artifactsDirectory.CombineWithFilePath($"{libmongocryptSolutionFile.GetFilename().ToString()}.{releaseVersion}.nupkg");
+        Information(packageFilePath);
+        NuGetPush(
+            packageFilePath,
+            new NuGetPushSettings 
+            {
+                ApiKey = nugetApi,
+                Source = "https://api.nuget.org/v3/index.json"
+            });
+        Information(gitVersion.FullSemVer.ToString());
+    });
+    
+Task("GreateGitTag")
+    .Does(() =>
+    {
+        ThrowIfLocalRelease();
+
+        Information("Show origin:");
+        Git(libmongocryptSolutionDirectory, "remote -v");
+        Git(libmongocryptSolutionDirectory, $"tag -a {csharpBindingsGitTagName} -m {csharpBindingsGitTagName}"); 
+        Git(libmongocryptSolutionDirectory, $"push origin {csharpBindingsGitTagName}"); 
+    });
+
+RunTarget(target);
+
+string GetSettingValue(string commandArgumentName, string defaultValue)
```

Comment:
this method allows us to get the required arguments from the initial command like:

> .\build.ps1 -ScriptArgs '--releaseVersion="1"'

or via the EnvironmentVariable with naming like: `LIBMONGOCRYPT_PACKAGING_RELEASEVERSION`